### PR TITLE
EPMRPP-78814 || [GA4] Previously saved parameters are sent after editing the rule

### PR DESCRIPTION
**Pull request by Bam6ycha**

### CODE REVIEW

**Review by chivekrodis — APPROVED** [review]

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/addEditNotificationModal.jsx [hunk 1]
```diff
@@ -241,32 +238,40 @@ const AddEditNotificationModal = ({
...
+      modalName,
+      status: informOwner,
+      type: messages[sendCase].defaultMessage,
+      switcher,
+      number: isEditorShown ? length : undefined,
+    };
 
-      trackEvent(PROJECT_SETTINGS_NOTIFICATIONS_EVENTS.CLICK_SAVE_BUTTON_IN_MODAL(eventParameters));
+    trackEvent(PROJECT_SETTINGS_NOTIFICATIONS_EVENTS.CLICK_SAVE_BUTTON_IN_MODAL(eventParameters));
+    onSave(!isEditorShown)(formFieldValues);
```
Comment (AmsterGet) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/addEditNotificationModal.jsx:265]: Seems like now there is no need to use high order function for `onSave` at `notifications.jsx`.
Please refactor it and simplify the call.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/addEditNotificationModal.jsx [hunk 2]
```diff
@@ -241,32 +238,45 @@ const AddEditNotificationModal = ({
...
-        messages[actionType].defaultMessage,
-      )}`;
+  const submitActions = (formFieldValues) => {
+    const modalName = `${messages.title.defaultMessage.replace(
+      /\{.*\}/i,
+      messages[actionType].defaultMessage,
+    )}`;
+
+    const newFormValues = { ...formFieldValues };
+
```
Comment (chivekrodis) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/addEditNotificationModal.jsx:250]: No needed spam with empty line if you define variables or functions. use it if you want split some peace of code. highlight something significant(usually if\else statements, return in the function, etc)
not a call to refactor this code

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/addEditNotificationModal.jsx [hunk 3]
```diff
@@ -241,32 +238,45 @@ const AddEditNotificationModal = ({
...
+    const {
+      informOwner,
+      sendCase,
+      enabled: switcher,
+      attributes: { length },
+    } = newFormValues;
 
-      handleSubmit(onSave(!isEditorShown))();
+    newFormValues.attributes = !isEditorShown ? [] : newFormValues.attributes;
```
Comment (chivekrodis) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/addEditNotificationModal.jsx:258]: You can do it directly when define `newFormValues`
const newFormValues = { ...formFieldValues , attributes: !isEditorShown ? [] : formFieldValues.attributes };

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx [hunk 4]
```diff
@@ -78,22 +78,18 @@ export const Notifications = ({ setHeaderTitleNode }) => {
...
 
-  const confirmEdit = (withoutAttributes) => (notification) => {
+  const confirmEdit = (notification) => {
     dispatch(
       updateProjectNotificationAction(
         convertNotificationCaseForSubmission({
           ...notification,
           name: notification.ruleName,
-          attributes: withoutAttributes ? [] : notification.attributes,
+          attributes: notification.attributes,
```
Comment (chivekrodis) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx:92]: Do you really need it here? You spread notification and right after that rewrite attribute from notification

**Review by chivekrodis — CHANGES_REQUESTED** [review]

**Review by chivekrodis — APPROVED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**